### PR TITLE
moving rest of the renderer unit tests over to pytest.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
moving rest of the renderer unit tests over to pytest.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by garethgreenaway** [comment]
@s0undt3ch wasn't sure how to best convert the function in the `test_stateconf`.py` test over to not run `RUNTIME_VARS`.

File: tests/pytests/unit/renderers/test_aws_kms.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+def test__cfg_data_key_no_key():
+    """
+    When no aws_kms:data_key is configured,
+    calling _cfg_data_key should raise a SaltConfigurationError
+    """
+    pytest.raises(salt.exceptions.SaltConfigurationError, aws_kms._cfg_data_key)
+
+
+def test__session_profile(aws_profile):  # pylint: disable=no-self-use
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_aws_kms.py:95]:
```suggestion
def test__session_profile(aws_profile):
```

File: tests/pytests/unit/renderers/test_aws_kms.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    """
+    _session raises a SaltConfigurationError
+    when boto3 raises botocore.exceptions.NoRegionError
+    """
+    with patch("boto3.Session") as session:
+        session.side_effect = botocore.exceptions.NoRegionError
+        pytest.raises(salt.exceptions.SaltConfigurationError, aws_kms._session)
+
+
+def test__kms():  # pylint: disable=no-self-use
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_aws_kms.py:125]:
```suggestion
def test__kms():
```

File: tests/pytests/unit/renderers/test_aws_kms.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    """
+    _kms raises a SaltConfigurationError
+    when boto3 raises a NoRegionError.
+    """
+    with patch("boto3.Session") as session:
+        session.side_effect = botocore.exceptions.NoRegionError
+        pytest.raises(salt.exceptions.SaltConfigurationError, aws_kms._kms)
+
+
+def test__api_decrypt(encrypted_data_key):  # pylint: disable=no-self-use
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_aws_kms.py:144]:
```suggestion
def test__api_decrypt(encrypted_data_key):
```

File: tests/pytests/unit/renderers/test_aws_kms.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    configured data key as the CiphertextBlob kwarg.
+    """
+    kms_client = MagicMock()
+    with patch.object(aws_kms, "_kms") as kms_getter:
+        kms_getter.return_value = kms_client
+        with patch.object(aws_kms, "_cfg_data_key", lambda: encrypted_data_key):
+            aws_kms._api_decrypt()
+            kms_client.decrypt.assert_called_with(
+                CiphertextBlob=encrypted_data_key
+            )  # pylint: disable=no-member
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_aws_kms.py:156]: This pylint dissable is really necessary?

File: tests/pytests/unit/renderers/test_stateconf.py [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+    return _renderers["stateconf"](
+        io.StringIO(content),
+        saltenv=saltenv,
+        sls=sls,
+        argline=argline,
+        renderers=salt.loader.render(config, {}),
+        **kws
+    )
+
+
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_stateconf.py:46]:
```suggestion
@attr.s
class Renderer:
    tmp_path = attr.ib()
    
    def __call__(self, content, sls="", saltenv="base", argline="-G yaml . jinja", **kws):
        state_tree_dir = self.tmp_path / "state_tree"
        cache_dir = self.tmp_path / "cachedir"
        state_tree_dir.mkdir()
        cache_dir.mkdir()
        config = salt.config.minion_config(None)
        config["root_dir"] = str(root_dir)
        config["state_events"] = False
        config["id"] = "match"
        config["file_client"] = "local"
        config["file_roots"] = dict(base=[str(state_tree_dir)])
        config["cachedir"] = str(cache_dir)
        config["test"] = False
        _renderers = salt.loader.render(config, {"config.get": lambda a, b: False})
        return _renderers["stateconf"](
            io.StringIO(content),
            saltenv=saltenv,
            sls=sls,
            argline=argline,
            renderers=salt.loader.render(config, {}),
            **kws
        )

@pytest.fixture
def renderer(tmp_path)
    return Renderer(tmp_path)
```

File: tests/pytests/unit/renderers/test_stateconf.py [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+def test_state_config():
+    result = _render_sls(
+        """
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_stateconf.py:49]:
```suggestion
def test_state_config(renderer):
    result = renderer(
        """
```
See where I'm getting at?

File: tests/pytests/unit/renderers/test_stateconf.py [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+
+""".format(
+                req
+            ),
+            sls="test",
+        )
+        assert len(result) == 2
+        assert "test::test" in result
+        assert "state_id" in result
+        assert result["state_id"]["cmd.run"][2][req][0]["cmd"] == "test::test"
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_stateconf.py:173]:
```suggestion
@pytest.mark.parametrize("req", REQUISITES)
def test_dot_state_id_in_requisites(req):
    result = _render_sls(
            """
.test:
  cmd.run:
    - name: echo renamed
    - cwd: /

state_id:
  cmd.run:
    - name: echo not renamed
    - cwd: /
    - {}:
      - cmd: .test

""".format(
                req
            ),
            sls="test",
        )
    assert len(result) == 2
    assert "test::test" in result
    assert "state_id" in result
    assert result["state_id"]["cmd.run"][2][req][0]["cmd"] == "test::test"
```

File: tests/pytests/unit/renderers/test_stateconf.py [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+
+
+def test_relative_include_with_requisites():
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_stateconf.py:176]: parametrize REQUISITES as suggested above

File: tests/pytests/unit/renderers/test_stateconf.py [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+  .utils::some_state:
+    cmd.run:
+      - name: echo overridden
+    """,
+        sls="test.work",
+    )
+    assert "test.utils::some_state" in result["extend"]
+
+
+def test_multilevel_relative_include_with_requisites():
```
Comment (s0undt3ch) [inline tests/pytests/unit/renderers/test_stateconf.py:218]: parametrize REQUISITES as suggested above

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
In addition to the requested changes, all of the `# pylint: disable`, make sure they're really necessary.

**Review by s0undt3ch — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**